openedx/core/djangoapps/credit/tasks.py

```diff
@@ -50,6 +41,101 @@ def update_course_requirements(course_id):
+    """
+    icrv_requirements = _get_credit_course_requirement_xblocks(course)
+    min_grade_requirement = _get_min_grade_requirement(course)
+    credit_requirements = icrv_requirements + min_grade_requirement
+    return credit_requirements
+
+
+def _get_min_grade_requirement(course):
+    """ Returns the list of minimum_grade_credit requirements for the given course
+
+    Args:
+        course(Course): The course object
+
+    Raises:
+        AttributeError if the course has not minimum_grade_credit attribute
+
+    Returns:
+        The list of minimum_grade_credit requirements
+    """
+    requirement = []
+    try:
+        requirement = [
+            {
+                "namespace": "grade",
+                "name": "grade",
+                "criteria": {
+                    "min_grade": getattr(course, "minimum_grade_credit")
+                }
+            }
+        ]
+    except AttributeError:
+        LOGGER.error("The course %s does not has minimum_grade_credit attribute", unicode(course.id))
+    return requirement
+
+
+def _get_credit_course_requirement_xblocks(course):  # pylint: disable=invalid-name
+    """ Generates a course structure dictionary for the specified course.
+
+    Args:
+        course(Course): The course object
+
+    Returns:
+        The list of credit requirements xblocks dicts
+    """
+
+    blocks_stack = [course]
+    requirements_blocks = []
+    while blocks_stack:
+        curr_block = blocks_stack.pop()
+        children = curr_block.get_children() if curr_block.has_children else []
+        if _is_credit_requirement(curr_block):
+            block = {
+                "namespace": curr_block.get_credit_requirement_namespace(),
+                "name": curr_block.get_credit_requirement_name(),
+                "criteria": ""
+            }
+            requirements_blocks.append(block)
+
+        # Add this blocks children to the stack so that we can traverse them as well.
+        blocks_stack.extend(children)
+    return requirements_blocks
+
+
+def _is_credit_requirement(xblock):
+    """ Check if the given xblock is a credit requirement
+
+    Args:
+        xblock(XBlock): The given xblock object
+
+    Returns:
+        True if xblock is a credit requirement else False
+    """
+
+    is_credit_requirement = False
+
+    if callable(getattr(xblock, "is_course_credit_requirement", None)):
+        is_credit_requirement = xblock.is_course_credit_requirement()
+        if is_credit_requirement:
+            if not callable(getattr(xblock, "get_credit_requirement_namespace", None)):
+                is_credit_requirement = False
+                LOGGER.error(
+                    "The has_namespace_function is not defined on xblock %s", xblock
+                )
+            if not callable(getattr(xblock, "get_credit_requirement_name", None)):
+                is_credit_requirement = False
+                LOGGER.error("The has_name_function is not defined on xblock %s", xblock)
```

Comment:
Please add a little bit more description to these error messages.  If people outside of our team see these errors, they should be able to tell immediately that this relates to credit eligibility requirements.

Maybe something like, "XBlock %v is marked as a credit requirement but does not implement `get_credit_requirement_name()`"

Also, please make sure that the method mentioned in the error message matches the method we're checking with `getattr`:
* has_name_function ==> get_credit_requirement_name
* has_namespace_function ==> get_credit_requirement_namespace